kroxylicious-test-tools/src/test/java/io/kroxylicious/test/tester/KroxyliciousTestersTest.java

```diff
@@ -124,24 +116,42 @@ void testSingleRequestClient(KafkaCluster cluster) {
             assertCanSendSingleRequestAndGetResponse(tester.singleRequestClient());
             assertCanSendSingleRequestAndGetResponse(tester.singleRequestClient(DEFAULT_VIRTUAL_CLUSTER));
         }
-        catch (Exception e) {
-            fail("unexpected exception", e);
-        }
     }
 
     @Test
-    void testMockTester() {
+    void testSingleRequestMockTester() {
         try (var tester = mockKafkaKroxyliciousTester(KroxyliciousConfigUtils::proxy)) {
             assertCanSendSingleRequestAndReceiveMockMessage(tester, tester.singleRequestClient());
             assertCanSendSingleRequestAndReceiveMockMessage(tester, tester.singleRequestClient(DEFAULT_VIRTUAL_CLUSTER));
         }
-        catch (Exception e) {
-            fail("unexpected exception", e);
+    }
+
+    @Test
+    void testMultiRequestMockTester() {
+        try (var tester = mockKafkaKroxyliciousTester(KroxyliciousConfigUtils::proxy)) {
+            assertCanSendMultipleRequestAndReceiveMockMessage(tester, tester.multiRequestClient());
+            assertCanSendMultipleRequestAndReceiveMockMessage(tester, tester.multiRequestClient(DEFAULT_VIRTUAL_CLUSTER));
         }
     }
 
     @Test
-    void testIllegalToAskForNonExistantVirtualCluster(KafkaCluster cluster) {
+    void testMultiRequestClientReportsConnectionState() {
```

Comment:
Should we have a version of the test that verifies the singleRequest client is closed after its request is complete.

Reply:
I got rid of the single request client.